lib/yard/templates/helpers/html_syntax_highlight_helper.rb

```diff
@@ -39,7 +39,7 @@ def html_syntax_highlight_ruby_ripper(source)
           end
           output
         rescue Parser::ParserSyntaxError
-          source =~ /^<span\s+class=/ ? source : h(source)
+          h(source)
```

Comment:
This is a subtle but problematic change that I think will cause a regression. This change was explicitly added to permit pre-formatted HTML extra-files to be added via: `yard - OTHER.html` where OTHER.html might include:

```html
<pre class="code ruby"><code class="ruby"><span class='kw'>return</span></code></pre>
```

![image](https://user-images.githubusercontent.com/686/171492756-c19d5812-2220-45c9-a623-1135cf932bcf.png)

The output of this section would generate double encoded output if you ran it through this block:

![image](https://user-images.githubusercontent.com/686/171492693-7f14148b-6f02-4d79-89e8-b9c81fc68690.png)

If removing this HTML detection causes a failure in your implementation, you may have to rethink part of your approach.